int32[4] botIds
int32[4] instructions
int32[4] forwardSpeeds
int32[4] rotationalSpeeds
